### PR TITLE
Use stackless errors

### PR DESCRIPTION
**Pull request by kriskowal**
r @yarpc/node

### CODE REVIEW

**Comment by Raynos** [comment]
There is quite a bit of nuance here.

You do not want stackless errors for everything.

 - You want real errors when an operational error occurs in the client
 - You want stack less "error frame" instances when you receive an error frame from a remote server.

Basically any error that is generated inside this process because of something that went wrong in this process should get a stack.

Any error that came in over the wire, aka tchannel error frame, aka HTTP 500, aka thrift exception, you want a stackless error.

File: lib/errors.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    source += 'properties = properties || {};\n';
+    source += 'this.message = this._format(this._template, properties);\n';
+    var names = Object.keys(properties);
+    for (var i = 0; i < names.length; i++) {
+        var name = names[i];
+        if (name === 'message') {
+            continue;
+        }
+        var value = properties[name];
+        source += '    this.' + name + ' = properties.' + name + ' || ' + JSON.stringify(value) + ';\n';
```
Comment (Matt-Esch) [inline lib/errors.js]: this of course only works if the property names are valid identifiers and not otherwise

Comment (kriskowal) [inline lib/errors.js]: Is this an acceptable limitation?

Comment (Raynos) [inline lib/errors.js]: the limitation is reasonable; i would add an regex assert though.

**Pull request closed without merging**